packages/jaeger-ui/src/components/TracePage/TracePageHeader/SpanGraph/index.tsx

```diff
@@ -51,37 +40,24 @@ function getItem(span: Span) {
   };
 }
 
-export default class SpanGraph extends React.PureComponent<SpanGraphProps, SpanGraphState> {
-  state: SpanGraphState;
+function getItems(trace: Trace) {
+  return trace.spans.map(getItem);
+}
 
+const memoizedGetItems = memoizeOne(getItems);
```

Comment:
+1

Reply:
How permanent is memoization? It looks like this will remember all traces visited by the user. Could it cause memory leak?

Reply:
This only memoized the last argument, if the next call of the function is called with the same arguments it won't execute the function again but returns the last results. If the call is with different arguments it will execute the function and cache only those last results.